RecoLocalCalo/HGCalRecProducers/plugins/HGCalLayerClustersFromSoAProducer.cc

```diff
@@ -87,8 +87,12 @@ class HGCalLayerClustersFromSoAProducer : public edm::stream::EDProducer<> {
     // each cluster. The size of 16 is a heuristic to minimize memory
     // reallocation and the associated overhead when the vector's capacity
     // needs to be extended.
-    std::vector<std::vector<float>> times(clusters->size(), std::vector<float>(16, 0.0f));
-    std::vector<std::vector<float>> timeErrors(clusters->size(), std::vector<float>(16, 0.0f));
+    std::vector<std::vector<float>> times(clusters->size());
+    std::vector<std::vector<float>> timeErrors(clusters->size());
+    for (unsigned int i = 0; i < clusters->size(); ++i) {
+      times[i].reserve(16);
+      timeErrors[i].reserve(16);
+    }
```

Comment:
Would it make sense to embed this loop into the previous one at lines 69 to 77? In this way, the heuristic is not even necessary and the size can be reserved correctly (i.e. the maximum possible, eventually some rechits will be discarded from the timing computation).